tests/integration/targets/aci_l3out/tasks/main.yml

```diff
@@ -0,0 +1,257 @@
+# Test code for the ACI modules
+# Copyright: (c) 2023, Gaspard Micol (@gmicol) <gmicol@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Test that we have an ACI APIC host, ACI username and ACI password
+  fail:
+    msg: 'Please define the following variables: aci_hostname, aci_username and aci_password.'
+  when: aci_hostname is not defined or aci_username is not defined or aci_password is not defined
+
+- name: Set vars
+  set_fact:
+    aci_info: &aci_info
+      host: "{{ aci_hostname }}"
+      username: "{{ aci_username }}"
+      password: "{{ aci_password }}"
+      validate_certs: '{{ aci_validate_certs | default(false) }}'
+      use_ssl: '{{ aci_use_ssl | default(true) }}'
+      use_proxy: '{{ aci_use_proxy | default(true) }}'
+      output_level: debug
+    
+# CLEAN ENVIRONMENT
+- name: Remove the ansible_tenant
+  aci_tenant:
+    <<: *aci_info 
+    tenant: ansible_tenant
+    state: absent
+
+- name: Verify Cloud and Non-Cloud Sites in use.
+  include_tasks: ../../../../../../integration/targets/aci_cloud_provider/tasks/main.yml
+
+- name: Execute tasks only for non-cloud sites
+  when: query_cloud.current == []  # This condition will execute only non-cloud sites
+  block:  # block specifies execution of tasks within, based on conditions
+  - name: Add a new tenant
+    aci_tenant:
+      <<: *aci_info 
+      tenant: ansible_tenant
+      description: Ansible tenant
+      state: present
+
+  - name: Remove the ansible_l3out
+    aci_l3out: &aci_l3out_absent
+      <<: *aci_info
+      tenant: ansible_tenant
+      name: ansible_l3out
+      description: L3Out for ansible_tenant tenant
+      domain: ansible_dom
+      vrf: ansible_vrf
+      state: absent
+
+  - name: Remove the second ansible_l3out
+    aci_l3out: &aci_l3out_2_absent
+      <<: *aci_info
+      tenant: ansible_tenant
+      name: ansible_l3out_2
+      description: Second L3Out
+      domain: ansible_dom
+      vrf: ansible_vrf
+      l3protocol: ospf
+      state: absent
+  
+  - name: Add a new L3Out (check_mode)
+    aci_l3out: &aci_l3out_present
+      <<: *aci_l3out_absent
+      state: present
+    check_mode: true
+    register: cm_add_l3out
+
+  - name: Add a new L3Out (normal_mode)
+    aci_l3out:
+      <<: *aci_l3out_present
+    register: nm_add_l3out
+  
+  - name: create L3Out again - testing idempotency
+    aci_l3out:
+      <<: *aci_l3out_present
+    register: create_idempotency
+
+  - name: Add the second ansible_l3out
+    aci_l3out:
+      <<: *aci_l3out_2_absent
+      state: present
+    register: cm_add_l3out_2
+
+  - name: asserts for l3out creation tasks
+    assert:
+      that:
+        - cm_add_l3out is changed
```

Comment:
can you add asserts to verify the check mode displays correctly? Would be something similar to below example

```
    - cm_span_src_1 is changed
    - cm_span_src_1.previous == []
    - cm_span_src_1.current == []
    - cm_span_src_1.proposed.spanSrc.attributes.name == "span_src_1"
    - cm_span_src_1.proposed.spanSrc.attributes.dn == "uni/infra/srcgrp-span_src_group_1/src-span_src_1"
    - cm_span_src_1.proposed.spanSrc.children.0.spanRsSrcToFilterGrp.attributes.tDn == "uni/infra/filtergrp-span_filter_group_1"
```

Reply:
yes, I will.